### PR TITLE
NRFU test case placeholders

### PR DESCRIPTION
**Pull request by vishwajeetgslab**
# Please include a summary of the changes

* nrfu_tests/master_def.yaml: Added placeholder for testcases. Which will keep the testcases in sequence in file also it will avoid the conflicts on another PR when one of the PR is merged.
* nrfu_tests/test_definition.yaml: Added placeholder for testcases. Which will keep the testcases in sequence in file also it will avoid the conflicts on another PR when one of the PR is merged.
* nrfu_tests/test_definition.yaml.j2: Added placeholder for testcases. Which will keep the testcases in sequence in file also it will avoid the conflicts on another PR when one of the PR is merged.
* pytest.ini: updated marker required for NRFU test cases

# Any specific logic/part of code you need extra attention on

Explain any specific logic you need special attention on

# Include the Issue number and link

Make sure to link the issue in the github PR UI 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (Testcase placeholder)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

**Review by MaheshGSLAB — APPROVED** [review]

File: nrfu_tests/master_def.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+common_data:
+  report_style: modern # If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
```
Comment (kgrozis-arista) [inline nrfu_tests/master_def.yaml:2]: report_style isn't needed in master def.  it should always be set to modern

File: nrfu_tests/master_def.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
+common_data:
```
Comment (kgrozis-arista) [inline nrfu_tests/master_def.yaml:1]: dut filter global parameter.  if it is not set nrfu test cases should default  to all duts.

File: nrfu_tests/master_def.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  report_style: modern # If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
+testcase_data:
```
Comment (kgrozis-arista) [inline nrfu_tests/master_def.yaml:3]: Most test cases will not have any parameters. We want to make nrfu testing very simple.  If there isn't a parameter we shouldn't have a placeholder.  Example if no parameters for NRFU1.1, we should not have mentions to the test case:
NRFU1.1: None
NRFU1.1: {}

File: nrfu_tests/test_definition.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+  {% set master_data = testcase_data %}
+  {% set report_style = common_data["report_style"] %}
+  testcases:
+    - name: test_
+      description:
+      test_id: NRFU1.1
+      {% set test_data = master_data["NRFU1.1"] %}
+      show_cmd: null
+      expected_output: null
+      report_style: {{ report_style }}
```
Comment (kgrozis-arista) [inline nrfu_tests/test_definition.yaml.j2:12]: Hard code this to modern

File: nrfu_tests/test_definition.yaml.j2 [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+      criteria: {{ test_data["criteria"] }}
```
Comment (kgrozis-arista) [inline nrfu_tests/test_definition.yaml.j2:13]: Set this value using global dut filter setting

**Review by kgrozis-arista — CHANGES_REQUESTED** [review]

File: nrfu_tests/master_def.yaml (hunk 1 above)
Comment (vishwajeetgslab) [reply nrfu_tests/master_def.yaml]: Removed report_style parameter from master definitions. Added modern as in j2 file.

File: nrfu_tests/master_def.yaml (hunk 2 above)
Comment (vishwajeetgslab) [reply nrfu_tests/master_def.yaml]: Add global_dut_filter as a parameter in master def.

File: nrfu_tests/master_def.yaml (hunk 3 above)
Comment (vishwajeetgslab) [reply nrfu_tests/master_def.yaml:5]: Updated as suggested.

File: nrfu_tests/test_definition.yaml.j2 (hunk 4 above)
Comment (vishwajeetgslab) [reply nrfu_tests/test_definition.yaml.j2]: Added modern directly in j2 file now. Thanks

File: nrfu_tests/test_definition.yaml.j2 (hunk 5 above)
Comment (vishwajeetgslab) [reply nrfu_tests/test_definition.yaml.j2]: Updated. Thanks

File: nrfu_tests/master_def.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+global_dut_filter:
```
Comment (kgrozis-arista) [inline nrfu_tests/master_def.yaml:1]: This stanza is optional.  If it is not provided by users than all test cases should be run on each dut.

**Review by geetanjalimanegslab — APPROVED** [review]

File: nrfu_tests/test_definition.yaml.j2 [hunk 7]
```diff
@@ -0,0 +1,274 @@
+- name: nrfu_tests
+  testcases:
+    - name: test_
+      description:
+      test_id: NRFU1.1
+      show_cmd: null
+      expected_output: null
+      report_style: modern
+      criteria: {{ global_dut_filter.criteria }}
```
Comment (kgrozis-arista) [inline nrfu_tests/test_definition.yaml.j2:9]: If no DUT filter is set then this will break because it will call variables that do not exist

**Review by kgrozis-arista — CHANGES_REQUESTED** [review]

**Review by shachiagarwal — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**